api/service.go

```diff
@@ -703,16 +902,21 @@ func (s *service) RemoveFile(ctx context.Context, req *pb.RemoveFileRequest) (*p
 	if !ok {
 		log.Fatal("scope required")
 	}
-	file, err := s.getFileWithScope(ctx, req.Path, scope)
+	folderName, fileName := parsePath(req.Path)
+	folder, err := s.getFolderWithScope(ctx, folderName, scope)
 	if err != nil {
 		return nil, err
 	}
+	folderPath := path.New(folder.Path)
 
-	if err = s.collections.Files.Delete(ctx, file.ID); err != nil {
+	dirpth, err := s.ipfsClient.Object().RmLink(ctx, folderPath, fileName)
```

Comment:
Similar to adding a file, remove the link and then move the pin to the new root.